### PR TITLE
✨ Add metamodeling load tests

### PR DESCRIPTION
**Pull request by bisgaard-itis**
## What do these changes do?
- refactor the performance tests
- update the version of `locust` we use
- Add a load test which runs the scenario from https://github.com/wvangeit/osparc-pyapi-tests using locust so it is easier to extend to other scenarios in the future. 

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by sanderegg — APPROVED** [review]
cool, and thanks for the update!

File: tests/performance/locust_files/.env-devel [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# user
+SC_USER_NAME=
+SC_PASSWORD=
+
+# locust settings
+LOCUST_HOST=http://10.43.103.193.nip.io:8005
```
Comment (pcrespov) [inline tests/performance/locust_files/.env-devel:6]: rm this ip... add localhost or something equivalent

File: tests/performance/locust_files/metamodeling/.env-devel [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# user
+OSPARC_API_KEY=
+OSPARC_API_SECRET=
+TEMPLATE_UUID=
+
+# locust settings
+LOCUST_HOST=http://10.43.103.193:9081
```
Comment (pcrespov) [inline tests/performance/locust_files/metamodeling/.env-devel:7]: same here

File: tests/performance/locust_files/metamodeling/.env-devel [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# user
+OSPARC_API_KEY=
```
Comment (pcrespov) [inline tests/performance/locust_files/metamodeling/.env-devel:2]: set these env vars here something like "replace-with=key" etc

File: tests/performance/locust_files/metamodeling/passer.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import json
+import os
+import pathlib as pl
+
+input_path = pl.Path(os.environ["INPUT_FOLDER"])
```
Comment (pcrespov) [inline tests/performance/locust_files/metamodeling/passer.py:5]:
```python

def main():
   # write here your stuff


if __name__ == "__main__":
   main()
```

File: tests/performance/locust_files/metamodeling/passer.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import json
+import os
+import pathlib as pl
```
Comment (pcrespov) [inline tests/performance/locust_files/metamodeling/passer.py:3]: MINOR: pl = Poland? ;-) why not simply `from pathlib import Path`?

File: tests/performance/locust_files/metamodeling/passer.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import json
```
Comment (pcrespov) [inline tests/performance/locust_files/metamodeling/passer.py:1]: Document a bit why we need these?

File: tests/performance/locust_files/metamodeling/workflow.py [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+
+# Perform the following setup in order to run this load test:
+# 1. Copy .env-devel to .env in this directory and add your osparc keys to .env.
+# 2. Construct a study **template** according to study_template.png. passer.py is the file next to this file.
+# 3. Setup the locust settings in the .env file (see https://docs.locust.io/en/stable/configuration.html#all-available-configuration-options)
+# run 'make test target=metamodeling/workflow.py' in your terminal and watch the magic happen 🤩
+
+
+class UserSettings(BaseSettings):
+    osparc_api_key: str = Field(default=...)
```
Comment (pcrespov) [inline tests/performance/locust_files/metamodeling/workflow.py:30]: Q: I wonder whether these settings will capture uppercase `osparc_api_key` , `osparc_api_secret` from `.env` file

File: tests/performance/Dockerfile [hunk 8]
```diff
@@ -6,6 +6,9 @@ FROM locustio/locust:${LOCUST_VERSION}
 RUN pip3 --version && \
   pip3 install \
   faker \
```
Comment (pcrespov) [inline tests/performance/Dockerfile:8]: MINOR: keep lists alphabetical when possible

File: tests/performance/locust_files/.env-devel [hunk 9]
```diff
@@ -0,0 +1,11 @@
+# user
+SC_USER_NAME=
```
Comment (pcrespov) [inline tests/performance/locust_files/.env-devel:2]: add dummy name&password?

File: tests/performance/locust_files/metamodeling/workflow.py [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+from urllib3 import PoolManager, Retry
+
+_MAX_WAIT_SECONDS: Final[int] = 60
+
+
+# Perform the following setup in order to run this load test:
+# 1. Copy .env-devel to .env in this directory and add your osparc keys to .env.
+# 2. Construct a study **template** according to study_template.png. passer.py is the file next to this file.
+# 3. Setup the locust settings in the .env file (see https://docs.locust.io/en/stable/configuration.html#all-available-configuration-options)
+# run 'make test target=metamodeling/workflow.py' in your terminal and watch the magic happen 🤩
```
Comment (pcrespov) [inline tests/performance/locust_files/metamodeling/workflow.py:26]: :-D

**Review by pcrespov — APPROVED** [review]
thx. Left some comments...

File: tests/performance/locust_files/metamodeling/workflow.py (hunk 7 above)
Comment (bisgaard-itis) [reply tests/performance/locust_files/metamodeling/workflow.py:30]: That works. I use them like that in the e2e client tests too.

File: tests/performance/Dockerfile (hunk 8 above)
Comment (bisgaard-itis) [reply tests/performance/Dockerfile:8]: 👍
done

File: tests/performance/locust_files/.env-devel (hunk 1 above)
Comment (bisgaard-itis) [reply tests/performance/locust_files/.env-devel:6]: I will use the convention of doing `replace-with-<name>` for all the fields the user must/should add.

File: tests/performance/locust_files/.env-devel (hunk 9 above)
Comment (bisgaard-itis) [reply tests/performance/locust_files/.env-devel:2]: good point

**Review by wvangeit — APPROVED** [review]
Looks good to me, thanks.

File: tests/performance/locust_files/metamodeling/passer.py (hunk 5 above)
Comment (bisgaard-itis) [reply tests/performance/locust_files/metamodeling/passer.py:3]: Sure, good point. I will fix

File: tests/performance/locust_files/metamodeling/passer.py (hunk 4 above)
Comment (bisgaard-itis) [reply tests/performance/locust_files/metamodeling/passer.py:5]: done. thanks

File: tests/performance/locust_files/metamodeling/.env-devel (hunk 2 above)
Comment (bisgaard-itis) [reply tests/performance/locust_files/metamodeling/.env-devel:7]: good point. done

File: tests/performance/locust_files/metamodeling/passer.py (hunk 6 above)
Comment (bisgaard-itis) [reply tests/performance/locust_files/metamodeling/passer.py:1]: OK, I will not do this further now. I think the main doc is in the locust file as well as the png I added to understand the setup

**Pull request merged**